### PR TITLE
Export a missing type and add type annotations

### PR DESCRIPTION
**Pull request by prprabhu-ms**
# Description

This PR address the [API review council feedback](https://apiview.dev/Assemblies/Review/fce877aea4a4400391f55e985f1d92b4/1b734e3549974121b9705a8f8bdb4ac6?diffRevisionId=ad28b2d974a747acb32f28539a10a2a6&doc=False&diffOnly=True) on this feature branch.

The changes are minimial:
- Add a previously missing type in the package exports.
- Add type annotations to newly added functions.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
  No new CHANGELOG entries beyond those already in this feature branch.
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/communication/azure-communication-chat/azure/communication/chat/_shared/models.py [hunk 1]
```diff
@@ -172,8 +172,7 @@ def __init__(self, user_id, **kwargs):
             self.raw_id = _microsoft_teams_user_raw_id(self)
 
 
-def _microsoft_teams_user_raw_id(identifier):
-    # type (MicrosoftTeamsUserIdentifier) -> str
+def _microsoft_teams_user_raw_id(identifier: MicrosoftTeamsUserIdentifier) -> str:
```
Comment (DominikMe) [inline sdk/communication/azure-communication-chat/azure/communication/chat/_shared/models.py:175]: Would we need to make the same change for phone number?

It still seems to be
```python
def _phone_number_raw_id(identifier):
    # type (PhoneNumberIdentifier) -> str
```

Comment (DominikMe) [reply sdk/communication/azure-communication-chat/azure/communication/chat/_shared/models.py:175]: (and then across all copies of the _shared folder)

**Review by DominikMe — APPROVED** [review]
One comment, otherwise lgtm

File: sdk/communication/azure-communication-sms/azure/communication/sms/_shared/models.py [hunk 2]
```diff
@@ -187,15 +186,14 @@ def _microsoft_teams_user_raw_id(identifier):
...
 
 
-def identifier_from_raw_id(raw_id):
+def identifier_from_raw_id(raw_id: str) -> CommunicationIdentifier:
     """
     Creates a CommunicationIdentifier from a given raw ID.
 
     When storing raw IDs use this function to restore the identifier that was encoded in the raw ID.
 
     :param raw ID to construct the CommunicationIdentifier from.
```
Comment (AikoBB) [inline sdk/communication/azure-communication-sms/azure/communication/sms/_shared/models.py:195]:
```suggestion
    :param str raw_id: A raw ID to construct the CommunicationIdentifier from
```
we should keep the reST docstring format, as it's used everywhere in this repo. Same suggestion for other occurrences in other acs modalities

File: sdk/communication/azure-communication-chat/azure/communication/chat/_shared/models.py (hunk 1 above)
Comment (prprabhu-ms) [reply sdk/communication/azure-communication-chat/azure/communication/chat/_shared/models.py:175]: It was a private function so I cared less about types vs comments (existing stuff in the file uses comments).
Changed now.

**Pull request merged**